stm32-modules/heater-shaker/simulator/sim_driver.cpp

```diff
@@ -0,0 +1,101 @@
+#include "simulator/sim_driver.hpp"
+
+#include <boost/asio.hpp>
+#include <regex>
+
+#include "simulator/simulator_queue.hpp"
+
+using namespace sim_driver;
+
+sim_driver::SocketSimDriver::SocketSimDriver(std::string url) {
+    std::regex url_regex(":\\/\\/([a-zA-Z0-9.]*):(\\d*)$");
+    std::smatch url_match_result;
+
+    if (std::regex_search(url, url_match_result, url_regex)) {
+        host = url_match_result[1];
+        port = std::stoi(url_match_result[2]);
+
+    } else {
+        std::cerr << "Malformed url." << std::endl;
+        exit(1);
+    }
+}
+
+std::string sim_driver::SocketSimDriver::get_host() { return this->host; }
+
+int sim_driver::SocketSimDriver::get_port() { return this->port; }
+
+std::string sim_driver::SocketSimDriver::get_name() { return this->name; }
+
+void sim_driver::SocketSimDriver::write() {}
+
+auto get_socket(std::string host, int port) {
+    boost::asio::io_service io_context;
+
+    boost::asio::ip::tcp::socket socket(io_context);
+    boost::asio::ip::tcp::endpoint endpoint(
+        boost::asio::ip::address::from_string(host), port);
+    boost::system::error_code ec;
+    socket.connect(
+        endpoint,
+        ec);  // TODO: Change this to socket.bind - Derek Maggio 9/30/21
+    if (ec) {
+        std::cerr << "Failed to create socket: " << ec.category().name() << ": "
+                  << ec.value() << std::endl;
+        exit(ec.value());
+    }
+    return socket;
+}
+
+void sim_driver::SocketSimDriver::read(
+    tasks::Tasks<SimulatorMessageQueue>& tasks) {
+    char pBuff[30];
+    auto socket = get_socket(this->host, this->port);
+    auto linebuf = std::string(1024, 'c');
```

Comment:
Data this big shouldn't be stored on the stack (allocated inline). This should at some point be created with a `new` (i.e., actually a `shared_ptr`).